### DISCUSSION TITLE
Doubt in MagickWand

### DISCUSSION THREAD

**Discussion by programmerarunkumar** [ImageMagick/ImageMagick#5029] [Help]
HI 

Im getting the below Exception 
MagickCore/magick.c:451: GetMagickDecoderThreadSupport: Assertion `magick_info->signature == MagickCoreSignature' failed.

MagickWandGenesis();
NewMagickWand();
MagickReadImage();
MagickResizeImage();
MagickWriteImages();
DestroyMagickWand();
MagickWandTerminus();

Im using the above mentioned MagickWand method to resize the image 
Multiple Thread will be calling those method at a time 

Can someone  share the possible reason for the above mentioned Exception ?

**Comment by snibgo** [comment]
What version of IM, on what platform?

> Can someone share the possible reason for the above mentioned Exception ?

You can read the code for yourself. In the file MagickCore/magick.c, function GetMagickDecoderThreadSupport(), there is a problem with magick_info->signature.

A search for GetMagickDecoderThreadSupport in the complete source code says this is called only by MagickCore\ReadImage(). So I suppose your program fails in the call to MagickReadImage().

I suspect a bug in your program or in IM. You haven't showed your program or an input image file, so I cannot reproduce the problem.
